pkg/sql/catalog/descs/hydrate.go

```diff
@@ -232,3 +233,45 @@ func HydrateGivenDescriptors(ctx context.Context, descs []catalog.Descriptor) er
 	}
 	return nil
 }
+
+// hydrateCatalog installs type metadata in the types present for all
+// table descriptors in a catalog.
+func hydrateCatalog(ctx context.Context, cat nstree.Catalog) error {
+	// Since we have a catalog, we already have everything we need to hydrate our
+	// types. Set up an accessor for the type hydration method to look into the
+	// catalog.
+	typeLookup := func(ctx context.Context, id descpb.ID) (tree.TypeName, catalog.TypeDescriptor, error) {
+		typDesc := cat.LookupDescriptorEntry(id)
+		if typDesc == nil {
+			n := tree.MakeUnresolvedName(fmt.Sprintf("[%d]", id))
+			return tree.TypeName{}, nil, sqlerrors.NewUndefinedObjectError(&n,
+				tree.TypeObject)
+		}
+		dbDesc := cat.LookupDescriptorEntry(typDesc.GetParentID())
+		if dbDesc == nil {
+			n := fmt.Sprintf("[%d]", typDesc.GetParentID())
+			return tree.TypeName{}, nil, sqlerrors.NewUndefinedDatabaseError(n)
+		}
+		scDesc := cat.LookupDescriptorEntry(typDesc.GetParentSchemaID())
+		name := tree.MakeQualifiedTypeName(dbDesc.GetName(), scDesc.GetName(), typDesc.GetName())
+		return name, typDesc.(catalog.TypeDescriptor), nil
```

Comment:
Can you check this type assertion and return an error if it is not a `TypeDescriptor`?

Reply:
Done.